.github/workflows/release.yml

```diff
@@ -0,0 +1,21 @@
+name: Release
```

Comment:
I do not know github actions well enough to know how to do this, but I think this workflow should depend on the ci workflow. We do not want to publish something that has failed tests.

Reply:
It will only run this workflow if it reaches `main` or a tag. Hopefully if it reaches `main` or a tag only if all tests passed in a PR first.

(this is the standard `release.yml` proposed by sbt-ci-release itself)

Reply:
What about:
- PR merge races?
- main having a larger matrix than the PR build?

I'm ok with proceeding like this, provided we only tag commits where we have manually verified the main passes.

I just don't fully grok the trade-off made here. If we block on CI, we have a slightly more complicated, slightly slower pipeline. But we avoid pushing a knowingly broken version.

Or is there another advantage to this that I'm missing?

Reply:
OK, fair enough. Then let's proceed with this (I guess the other option would be to duplicate / factor out `ci.yml`, but that's arguably overkill).

Reply:
> It doesn't seem like there is a good way to do this.

The [scala-js-macrotask-executor CI](https://github.com/scala-js/scala-js-macrotask-executor/blob/8bdd3c144eb4ea0f4aa2c7501315166fef6d9fa4/.github/workflows/ci.yml) is an example how to set this up.